modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,368 @@
+  sequenceId(nonce: sequenceId): this {
+    const value = new BigNumber(nonce.value);
+    this.validateValue(value);
+    this._nonce = value.toNumber();
+    return this;
+  }
+
+  /**
+   * The tip to increase transaction priority.
+   *
+   * @param {number | undefined} tip optional
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  tip(tip: FeeOptions): this {
+    const tipBN = new BigNumber(tip.amount);
+    this.validateValue(tipBN);
+    this._tip = tipBN.toNumber();
+    return this;
+  }
+
+  /**
+   * The number of the checkpoint block after which the transaction is valid
+   *
+   * @param {number} blockNumber
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  validity({ firstValid, maxDuration }: validityWindow): this {
+    if (firstValid) {
+      this.validateValue(new BigNumber(firstValid));
+      this._blockNumber = firstValid;
+    }
+    if (maxDuration) {
+      this.validateValue(new BigNumber(maxDuration));
+      this._eraPeriod = maxDuration;
+    }
+    return this;
+  }
+
+  /**
+   * The hash of the checkpoint block.
+   *
+   * @param {number} blockHash
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   * @see https://wiki.polkadot.network/docs/build-protocol-info#transaction-mortality
+   */
+  blockHash(blockHash: string): this {
+    this._blockHash = blockHash;
+    return this;
   }
 
   /**
-   * Initialize the transaction builder fields using the decoded transaction data
+   * The current version for transaction format.
+   *
+   * @param {number} transactionVersion
+   * @returns {TransactionBuilder} This transaction builder.
    *
-   * @param {Transaction} tx the transaction data
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  version(transactionVersion: number): this {
+    this._transactionVersion = transactionVersion;
+    return this;
+  }
+
+  private method(method: TxMethod): this {
+    this._method = method;
+    return this;
+  }
+
+  /**
+   * Set the network based on the configuration in the statics module
    */
-  initBuilder(tx: Transaction): void {
-    throw new NotImplementedError('initBuilder not implemented');
+  protected staticsConfig(): void {
+    const networkConfig = this._coinConfig.network as DotNetwork;
+    const { specName, genesisHash, specVersion, chainName, metadataRpc } = networkConfig;
+    this._genesisHash = genesisHash;
+    this._specVersion = specVersion;
+    this._chainName = chainName;
+    this._specName = specName;
+    this._metadataRpc = metadataRpc;
+    this._registry = getRegistry({
+      chainName: chainName,
+      specName: specName,
+      specVersion: specVersion,
+      metadataRpc: metadataRpc,
+    });
```

Comment:
I see that statics values are getting from network config. Testnet method was removed to enforce the relationship between coins and network. The modular division of this project is being respected.

But my understanding of polkadot is not enough to say how things are.  I don't know the semantics of this values. Now I don't see a way to use account lib in dev env. Are they really static?
Why does [sidecar have a endpoint to get them](https://paritytech.github.io/substrate-api-sidecar/dist/#operations-transaction-getTransactionMaterial)?
How are we going to integrate this with the [data that we are making available in IMS](https://github.com/BitGo/indexerdb-microservice/blob/master/src/app/controllers/jsonrpc/dot.js#L22)?
Why is there another [staticsConfig in transactionBuilderFactory](https://github.com/BitGo/BitGoJS/blob/feat/dot-tx-builder-factory/modules/account-lib/src/coin/dot/transactionBuilderFactory.ts#L44)?

Reply:
@jrafael-simtlix you can override this configuration for a dev environment, or any environment using this appraoch:
https://github.com/BitGo/BitGoJS/blob/1864ed9157199cf167b90cc965470f5fe786cf17/modules/account-lib/test/unit/coin/dot/transactionBuilder/base.ts#L23

The configuration is modified to support transactions based on a dev node i setup locally. Likewise you could update the configuration to however you need based on this approach.

Reply:
The reason for it being in the factory is because we need this config to decode hex transactions